--- a/htdocs/luci-static/resources/view/firewall/zones-sonic.js
+++ b/htdocs/luci-static/resources/view/firewall/zones-sonic.js
@@ -6,6 +6,7 @@
 'require network';
 'require firewall';
 'require tools.firewall as fwtool';
+'require tools.fullconenat-sonic as fullcone';
 'require tools.widgets as widgets';
 
 return view.extend({
@@ -35,6 +36,7 @@ return view.extend({
 
 		m = new form.Map('firewall', _('Firewall - Zone Settings'),
 			_('The firewall creates zones over your network interfaces to control network traffic flow.'));
+		fullcone.attach(m);
 
 		s = m.section(form.TypedSection, 'defaults', _('General Settings'));
 		s.anonymous = true;
@@ -55,11 +57,7 @@ return view.extend({
 		};
 
 		o = s.option(form.Flag, 'drop_invalid', _('Drop invalid packets'));
-		if (L.hasSystemFeature('fullcone')) {
-			o = s.option(form.Flag, 'fullcone', _('Enable FullCone NAT'));
-			if (fw4)
-				o = s.option(form.Flag, 'fullcone6', _('Enable FullCone NAT6'));
-		}
+		fullcone.addGlobal(s);
 
 		let p = [
 			s.option(form.ListValue, 'input', _('Input', 'firewall input')),
@@ -184,6 +182,8 @@ return view.extend({
 			return null;
 		};
 
+		fullcone.addZone(s);
+
 		o = s.taboption('general', form.Flag, 'mtu_fix', _('MSS clamping'));
 		o.modalonly = true;
 
