--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -28,6 +28,7 @@
 #include <linux/of_address.h>
 #include <linux/of_mdio.h>
 #include <linux/phy.h>
+#include <linux/hash.h>
 
 #if defined(NSS_DP_PPE_SUPPORT)
 #include <fal/fal_vsi.h>
@@ -526,7 +527,7 @@ static int nss_dp_rx_flow_steer(struct net_device *netdev, void *_skb,
 	}
 
 	hash = skb_get_hash(skb);
-	rxflow = &flow_table->flows[hash & flow_table->mask];
+	rxflow = &flow_table->flows[hash_32(hash, flow_table->log)];
 	rxcpu = (uint32_t)rxflow->cpu;
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
@@ -1093,11 +1094,7 @@ MODULE_DEVICE_TABLE(of, nss_dp_dt_ids);
 static struct platform_driver nss_dp_drv = {
 	.probe = nss_dp_probe,
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
-	.remove_new = nss_dp_remove,
-#else
 	.remove = nss_dp_remove,
-#endif
 	.driver = {
 		   .name = "nss-dp",
 		   .owner = THIS_MODULE,
